From 735775241b770131f71135de0f7a7f4bc15da393 Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Wed, 9 Nov 2022 09:07:21 -0300
Subject: [PATCH] STX: Customize Debian build files

Adds to libvirt Debian build instructions, starlingx specific files
based on how starlingx-staging/stx-libvirt was previously built and
delivered on CentOS, adding to the following:
  * Enabling syslog for libvirtd
  * Setting "auth_tcp" to "none"
  * Logrotate files
  * Qemu hook

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
[ Fixed function renaming while porting patch to libvirt v8.0.0-1 ]
Signed-off-by: Murillo Arantes <murillo.arantes@windriver.com>
[ Ported to libvirt v11.3.0-3+deb13u2 ]
Signed-off-by: Murillo Arantes <murillo.arantes@windriver.com>
---
 debian/libvirt-daemon.install.in |  1 +
 debian/rules                     | 23 +++++++++++++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/debian/libvirt-daemon.install.in b/debian/libvirt-daemon.install.in
index 2bee02b8c..bce473693 100644
--- a/debian/libvirt-daemon.install.in
+++ b/debian/libvirt-daemon.install.in
@@ -1,5 +1,6 @@
 etc/apparmor.d/usr.sbin.libvirtd
 etc/libvirt/libvirtd.conf
+etc/libvirt/hooks/qemu
 etc/logrotate.d/libvirtd
 usr/lib/libvirt/virt-aa-helper
 usr/lib/systemd/system/libvirtd-admin.socket
diff --git a/debian/rules b/debian/rules
index 868b31876..566c6aaf8 100755
--- a/debian/rules
+++ b/debian/rules
@@ -18,6 +18,11 @@ export DPKG_GENSYMBOLS_CHECK_LEVEL
 
 include debian/arches.mk
 
+SOURCE1                   = libvirt.logrotate
+SOURCE2                   = libvirt.lxc
+SOURCE3                   = libvirt.qemu
+SOURCE4                   = qemu
+
 ifeq ($(DEB_HOST_ARCH_OS), linux)
     WITH_DAEMONS              = -Ddriver_libvirtd=enabled -Dhost_validate=enabled -Dssh_proxy=enabled
     WITH_ESX                  = -Ddriver_esx=enabled -Dcurl=enabled
@@ -275,6 +280,24 @@ ifeq ($(DEB_HOST_ARCH_OS), linux)
 	mkdir -p $(DEB_DESTDIR)/etc/apt/apt.conf.d/
 	cp debian/apt/* \
 	   $(DEB_DESTDIR)/etc/apt/apt.conf.d/
+
+	# STX: Begin custom install
+	# Enable syslog for libvirtd (/var/log/libvirtd.log)
+	echo "log_outputs=\"3:syslog:libvirtd\"" >> $(DEB_DESTDIR)/etc/libvirt/libvirtd.conf
+
+	# Set auth_tcp to "none" for now to enable live migration.
+	# We'll need to set up proper authentication later.
+	sed -i '/#auth_tcp/a auth_tcp = "none"' $(DEB_DESTDIR)/etc/libvirt/libvirtd.conf
+
+	# Install logrotate files
+	install -p -D -m 644 $(SOURCE1) $(DEB_DESTDIR)/etc/logrotate.d/libvirtd
+	install -p -D -m 644 $(SOURCE2) $(DEB_DESTDIR)/etc/logrotate.d/libvirtd.lxc
+	install -p -D -m 644 $(SOURCE3) $(DEB_DESTDIR)/etc/logrotate.d/libvirtd.qemu
+
+	# Install hooks
+	mkdir -p $(DEB_DESTDIR)/etc/libvirt/hooks
+	install -m 0500 $(SOURCE4) $(DEB_DESTDIR)/etc/libvirt/hooks/qemu
+	# STX: End custom install
 endif
 
 	# Copy the release notes where dh_installdocs can find them
-- 
2.34.1

